Rebuild travel and lifestyle page

mtruj013 opened the pull request:
## Done

- Applied redesign as per [design](https://www.figma.com/file/KQSCmTqOjY4gfKvJPyvzj2/Canonical.com-latest-only?node-id=7367%3A20953&t=Ghb8qeVFsNnSVkNO-0)

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: https://canonical-com-810.demos.haus/careers/travel
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Check against redesign
- Note that the copy for this page is a combination of what were once 2 different pages: [travel](https://docs.google.com/document/d/1H2ILhAUckOWyah_6V69ImwLWOSRlXkaYnIhOHkUjclc/edit#heading=h.jyjo9jkeh22g) and [lifestyle](https://docs.google.com/document/d/15Ofb6vRUXbQ3evcWmrOu9ymuY37ayg4M_xZ2QXQyIts/edit#heading=h.jyjo9jkeh22g). So please check the final copy against how these were merged in the design

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-177

webteam-app commented:
Demo starting at https://canonical-com-810.demos.haus

petesfrench commented on static/js/sprint-map.js:
```diff
@@ -0,0 +1,81 @@
+    [{ lat: 30.355265001335628, lng: -97.76687845245245 }, "Austin"],
+    [{ lat: 48.857074990278676, lng: 2.3395547447652056 }, "Paris"],
+    [{ lat: 29.767193226058257, lng: -95.3670538129764 }, "Houston"],
+    [{ lat: -31.043359293503908, lng: 25.51255683173218 }, "Spier"],
```
I couldn't find this one on google maps

petesfrench commented on static/js/sprint-map.js:
```diff
@@ -0,0 +1,81 @@
+    [{ lat: 25.763155578558653, lng: 54.94246340504953 }, "Dubai"],
+    [{ lat: 47.65006756574396, lng: -122.35867431899362 }, "Seattle"],
+    [{ lat: 37.42579354504801, lng: -78.17007396261319 }, "Virginia"],
+    [{ lat: 41.078424184612594, lng: -9.051132205419249 }, "Portugal"],
```
This pin looks a little off
![image](https://user-images.githubusercontent.com/58276363/223736050-6e4627a5-8d41-417c-baef-b1ca13beb6ae.png)

petesfrench commented:
We should add a fallback for people who don't have JS enabled:
![image](https://user-images.githubusercontent.com/58276363/223736580-8afa0781-9aed-4720-9afb-db397efbb4ce.png)

anthonydillon commented:
I am going on a limb and adding a design +1 as the page seems to match the Figma design. Checked on different viewports and a11y reviewed.

anthonydillon commented:
There are some a11y issues introduced by Google maps which do not seem fixable

petesfrench commented on templates/careers/travel.html:
```diff
@@ -1,66 +1,112 @@
+          {{ 
+            image (
+            url="https://assets.ubuntu.com/v1/379b19b9-department-engineering-%402x.jpg",
+            alt="",
```
An alt text would be useful here

petesfrench commented on templates/careers/travel.html:
```diff
@@ -1,66 +1,112 @@
+    </div>
+  </div>  
+</section>
+<script defer src="{{ versioned_static('js/sprint-map.js') }}"></script>
```
Do you think it is worth adding this to the build?

mtruj013 replied:
I'm actually not sure. My first instinct was no because this isn't building something based on a dependency (like fuse js or latest news for example), but there's a very high likelihood that I'm totally wrong in that assessment 😅 @anthonydillon what would be best practice here?

marinacastejon commented:
@mtruj013 LGTM, thanks

mtruj013 commented:
@anthonydillon @petesfrench @marinacastejon  Thanks all for reviewing! I think I've addressed most concerns but I have some questions.

**JS not enabled**
I've added what I think is a very hacky fix for when JS is not enabled and I'd like to get your feedback. I moved the styles to be added within the script instead so that if JS is not enabled the div doesn't take a defined width and height. That fixed the huge empty div situation. But now the question is, is this the behavior we want? Would we rather have a static image shown as opposed to showing nothing in place of the map? 

**Image alt text**
For the image with 4 locations I wasn't sure what to set as the alt text as I don't actually know what cities are in the image. I can see NYC, Seoul, and what looks like Budapest but I've no idea about the other one. Is "Sprint locations" good enough here? 

**Nested strips**
This point was raised in MM, but just to clarify, the nested p-strip convention comes from design and is on every single department page. If this is not best practice then we should clarify and document that.

anthonydillon commented:
Thanks for all the work on this @mtruj013. 

**JS not enabled**
I would suggest we could just render a list of locations in the markup for non-js in a `<nosrcipt>`.

**Image alt text**
It would be better to describe the image the best you can. You could even use something like "An image split in four showing four cities that we have visited, including NYC, Seoul, etc".

**Nested strips**
Sounds good, I will bring that up with Vanilla for guidance.

**a11y bugs**
Wave is complaining about the markers not having an alt. After a quick Google it looks like there is no quick fix.

tbille commented:
- Mobile view of the map has a fixed width which makes the scroll vertically :scream: 
![image](https://user-images.githubusercontent.com/2707508/224272860-9472b9d4-41dd-4044-ae98-e21dce1bfa28.png)

- Same issue with JS disabled:
![image](https://user-images.githubusercontent.com/2707508/224273318-2861399a-55c1-466c-90d8-c3b3b3484df5.png)

tbille commented on static/js/sprint-map.js:
```diff
@@ -0,0 +1,32 @@
+  
+  const mapDiv = document.getElementById("sprint-map");
+  mapDiv.style.height = "25rem";
+  // mapDiv.style.width = "69rem";
```
This line can be removed.

carkod reviewed (approved):
LGTM
